src/client/mesh.cpp

```diff
@@ -328,6 +328,22 @@ void recalculateBoundingBox(scene::IMesh *src_mesh)
 	src_mesh->setBoundingBox(bbox);
 }
 
+bool checkMeshNormals(scene::IMesh *mesh)
+{
+	u32 buffersCount = mesh->getMeshBufferCount();
+
+	for (u32 i = 0; i < buffersCount; i++) {
+		scene::IMeshBuffer *buffer = mesh->getMeshBuffer(i);
+		core::vector3df firstNormal = buffer->getNormal(0);
+		f32 length = firstNormal.getLength();
+
+		if (length == 0 || length == INFINITY || length == NAN)
```

Comment:
aren't NaN comparisons always false?
also I've never seen those constants used anywhere, so this should just be `isinf` and `isnan`

Reply:
My bad, NaNs are really incomparable.

Reply:
Fixed